modules/core/src/test/java/org/apache/ignite/client/IgniteBinaryTest.java

```diff
@@ -123,6 +130,59 @@ public void testBinaryObjectPutGet() throws Exception {
         }
     }
 
+    /**
+     * Tests that {@code org.apache.ignite.cache.CacheInterceptor#onBeforePut(javax.cache.Cache.Entry, java.lang.Object)}
+     * throws correct exception in case while cache operations are called from thin client. Only BinaryObject`s are
+     * acceptable in this case.
+     */
+    @Test
+    public void testBinaryWithNotGenericInterceptor() throws Exception {
+        IgniteConfiguration ccfg = Config.getServerConfiguration()
+            .setCacheConfiguration(new CacheConfiguration("test").setInterceptor(new ThinBinaryValueInterceptor()));
+
+        String castErr = "cannot be cast to";
+        String treeErr = "Tree is corrupted";
+
+        ListeningTestLogger srvLog = new ListeningTestLogger(log);
+
+        LogListener lsnrCast = LogListener.matches(castErr).
+            andMatches(str -> !str.contains(treeErr)).build();
+
+        srvLog.registerListener(lsnrCast);
+
+        ccfg.setGridLogger(srvLog);
+
+        try (Ignite ignored = Ignition.start(ccfg)) {
+            try (IgniteClient client =
+                     Ignition.startClient(new ClientConfiguration().setAddresses(Config.SERVER))
+            ) {
+                ClientCache<Integer, ThinBinaryValue> cache = client.cache("test");
+
+                try {
+                    cache.put(1, new ThinBinaryValue());
```

Comment:
When there is no exception, this test won't fail and won't check anything.
Please either insert `assert false`, or use one of the `GridTestUtils.assertThrows` helpers.